app/controllers/api/base_controller.rb

```diff
@@ -38,17 +38,20 @@ def get_resource
 
     # overwrites resource_scope in FindCommon to consider nested objects
     def resource_scope(options = {})
-      options[:association] ||= controller_name
-      if nested_obj && nested_obj.respond_to?(options[:association])
-        association = nested_obj.send(options[:association])
-        if association.respond_to?(:authorized)
-          association.authorized(options[:permission], resource_class)
-        else
-          association
-        end
-      else
-        super(options)
-      end
+      child_scope = super(options)
+      child_scope.merge(parent_scope).readonly(false)
+    end
+
+    def parent_scope(options = {})
+      parent_name, parent_resource_scope = parent_resource_details()
+
+      return resource_class.where(nil) unless parent_resource_scope
```

Comment:
me thinks where({}) would be margianlly less horrible
![image](https://cloud.githubusercontent.com/assets/706359/6822095/b848494e-d2e5-11e4-8ea6-a339a132c12e.png)